arch/Kconfig

```diff
@@ -33,6 +33,13 @@ config RISCV32
 config XTENSA
 	bool "Xtensa architecture"
 
+config ARCH_POSIX
```

Comment:
I wonder if we should instead build a tree here:
```
ARCH_NATIVE
|-- ARCH_POSIX
|---- SOC_POSIX
|------ SOC_POSIX_MACOS
|------ SOC_POSIX_MACOS
|-- ARCH_WINX
|---- SOC_WIN32
...
```

Reply:
@galak thoughts?

Reply:
This port arch/soc & board is meant to be both for Unix/Linux and MacOS (I haven't tried it on MacOS, but everything I wrote should hopefully be POSIX compliant, including the simple_process board)
Regarding Windows, its threading is quite different than pthreads. So I guess if somebody wants that, they would need to build an ARCH_WIN and so forth.
Nesting ARCH_POSIX and a possible ARCH_WIN below an ARCH_NATIVE would seem to be about categorizing without a functional need. Also the current source tree does not seem to support it.

Reply:
I'm not quite convinced about Windows, since it's not only the threading, but also the build system and other basic changes that will be common to all native builds. That said I can't really hold the argument without actually trying a Windows native port, so I will let the others voice their opinion.